emp/emp_role_cftemplate.yaml

```diff
@@ -0,0 +1,377 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Parameters:
+  RoleName:
+    Type: String
+    Description: The name of the IAM role and instance profile.
+    Default: 'pf9emp'
+
+Resources:
+  # CAPA Policies
+  AWSIAMManagedPolicyCloudProviderNodes:
+    Properties:
+      Description: For the Kubernetes Cloud Provider AWS nodes
+      ManagedPolicyName: !Sub "${RoleName}-nodes.cluster-api-provider-aws.sigs.k8s.io"
+      PolicyDocument:
+        Statement:
+          - Action:
+              - ec2:AssignIpv6Addresses
+              - ec2:DescribeInstances
+              - ec2:DescribeRegions
+              - ec2:CreateTags
+              - ec2:DescribeTags
+              - ec2:DescribeNetworkInterfaces
+#              - ec2:DescribeInstanceTypes commenting, not deleting capa actions. This is accounted for below
+              # needed to pull an image from ecr. EKS bootstrap script adds an entry for the sandbox image
+              # with an ecr URL in containerd config, so these are needed
+              - ecr:GetAuthorizationToken
+              - ecr:BatchCheckLayerAvailability
+              - ecr:GetDownloadUrlForLayer
+              - ecr:GetRepositoryPolicy
+              - ecr:DescribeRepositories
+              - ecr:ListImages
+              - ecr:BatchGetImage
+            Effect: Allow
+            Resource:
+              - '*'
+          - Action:
+              # capa creates encrypted cloud init secrets, this is needed.
+              - secretsmanager:DeleteSecret
+              - secretsmanager:GetSecretValue
+            Effect: Allow
+            Resource:
+              - arn:*:secretsmanager:*:*:secret:aws.cluster.x-k8s.io/*
+          - Action:
+              # they are related to some heartbeat sent by systems manager see: https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-setting-up-messageAPIs.html
+              - ssm:UpdateInstanceInformation
+              - ssmmessages:CreateControlChannel
+              - ssmmessages:CreateDataChannel
+              - ssmmessages:OpenControlChannel
+              - ssmmessages:OpenDataChannel
+              # discuss why this is needed
```

Comment:
need to remove?

Reply:
its needed by capi :)

Reply:
OK, my only concern is as we put cft this to public folks, these commented actions will be confusing to users. As he is not aware CAPI or Qbert. He is only accounted for EMP Cloud formation template.

Reply:
Do we plan to update this policy before 1.1 again to remove all commented part and make it clean?